serverless.yml

```diff
@@ -7,22 +7,95 @@ plugins:
   - serverless-step-functions
   - serverless-pseudo-parameters
 
+custom:
+  DYNAMO_DB_TABLE: stackJanitorTable
+  DEFAULT_EXPIRATION_PERIOD: 604800 #seconds
+
 provider:
   name: aws
   runtime: nodejs10.x
+  memorySize: 128
+  stage: ${opt:stage, 'dev'}
   region: ap-southeast-2
   iamRoleStatements:
+    - Effect: "Allow"
+      Action:
+        - "dynamodb:DescribeTable"
+        - "dynamodb:Query"
+        - "dynamodb:Scan"
+        - "dynamodb:GetItem"
+        - "dynamodb:PutItem"
+        - "dynamodb:UpdateItem"
+        - "dynamodb:DeleteItem"
+      Resource: "arn:aws:dynamodb:*:*:table/${self:custom.DYNAMO_DB_TABLE}"
     - Effect: "Allow"
       Action:
         - "cloudformation:DescribeStack*"
+        - "cloudformation:DeleteStack*"
+      Resource: "*"
+    - Effect: "Allow"
+      Action:
+        - "ec2:*"
+        - "s3:*"
+        - "ecs:*"
+        - "elasticloadbalancing:*"
+        - "elasticloadbalancing:*"
+        - "elasticloadbalancingv2:*"
+        - "logs:*"
+        - "events:*"
+        - "route53:*"
+        - "cloudwatch:*"
+        - "application-autoscaling:*"
+        - "autoscaling:*"
+        - "apigateway:*"
       Resource: "*"
 
+resources:
+  Resources:
+    stackJanitorTable:
+      Type: AWS::DynamoDB::Table
+      Properties:
+        TableName: ${self:custom.DYNAMO_DB_TABLE}
+        AttributeDefinitions:
+          - AttributeName: stackName
+            AttributeType: S
+          - AttributeName: stackId
+            AttributeType: S
+
+        KeySchema:
+          - AttributeName: stackName
+            KeyType: HASH
+          - AttributeName: stackId
+            KeyType: RANGE
+
+        TimeToLiveSpecification:
+          AttributeName: expirationTime
+          Enabled: true
+
+        ProvisionedThroughput:
+          ReadCapacityUnits: 1
```

Comment:
Does it make sense to have 1 unit? Should we be using on-demand, or is current throughput small enough?

Reply:
Fixed.